src/Microsoft.OData.Core/ODataReaderCore.cs

```diff
@@ -368,7 +382,7 @@ public override sealed bool Read()
         public override sealed Task<bool> ReadAsync()
         {
             this.VerifyCanRead(false);
-            return this.ReadAsynchronously().FollowOnFaultWith(t => this.EnterScope(new Scope(ODataReaderState.Exception, null, null, null, null)));
+            return this.ReadAsynchronously().FollowOnFaultWith(t => this.EnterScope(new Scope(ODataReaderState.Exception, null, null,(IEdmStructuredTypeReference) null, null)));

```

Comment:
>IEdmStructuredTypeReference [](start = 134, length = 27)

I don't like to call using (IEdmStructuredTypeReference)null. It maybe a design refactor.